tests/index.test.js

```diff
@@ -56,14 +62,102 @@ describe('Test that the basic routes return dummy data', () => {
     }])
   })
 
-  test('POST /tasks/volunteer', async () => {
-    return supertest(await app(config))
-      .post('/tasks/volunteer')
+  test('POST /users/id/tasks/claim', async () => {
+    const config = loadConfig()
+    const provider = new MemoryUserProvider()
+    config.users.provider.provider = provider
+
+    await supertest(await app(config))
+      .post('/users/example@example.com/tasks/unclaim')
```

Comment:
Some characters in email addresses have special encoding or meaning in URLs, like `@` and `+`. I think we should either have thorough testing of how Express and our code handle these characters in the URL or make the email address part of the JSON body of the POST request.

Do you have a preference or another idea for handling email addresses?

Reply:
I do not have strong preferences. At the moment, this appears to work, but we may want to investigate. According to [this](https://stackoverflow.com/questions/19509028/can-i-use-an-at-symbol-inside-urls) and [this](https://tools.ietf.org/html/std66#section-3.3) it appears that `@` characters are explicitly allowed in the path part of URIs. However, I agree that this may cause problems. However, we would want to weigh the risk of there being issues with that against the usability of just being able to drop an email address or phone number directly into the url.

I'd be inclined to merge this with the current implementation and make a ticket for further investigation and future improvement, but if you have a strong preference toward using a url encoding that would be fine too.

Reply:
Works for me. 😁